src/deploy/functions/release/fabricator.ts

```diff
@@ -295,6 +303,36 @@ export class Fabricator {
         .catch(rethrowAs(endpoint, "create topic"));
     }
 
+    // Like Pub/Sub, GCF requires a channel to exist before allowing the function
+    // to be created. Like Pub/Sub we currently only support setting the name
+    // of a channel, so we can do this once during createFunction alone. But if
+    // Eventarc adds new features that we indulge in (e.g. 2P event providers)
+    // things will get much more complicated. We'll have to make sure we keep
+    // up to date on updates, and we will also have to worry about channels leftover
+    // after deletion possibly incurring bills due to events still being sent.
+    const channel = apiFunction.eventTrigger?.channel;
+    if (channel) {
+      await this.executor
+        .run(async () => {
+          try {
+            const op: { name: string } = await eventarc.createChannel({ name: channel });
+            return await poller.pollOperation<eventarc.Channel>({
+              ...eventarcPollerOptions,
+              pollerName: `create-${channel}-${endpoint.region}-${endpoint.id}`,
+              operationResourceName: op.name,
+            });
+          } catch (err: any) {
+            if (err.status === 409) {
```

Comment:
Can we add a comment that we get a CONFLICT because the channel has been created already and therefore we don't need to worry about the channel already existing?

Reply:
Re: channel normalization, I believe the normalization happens here: https://github.com/firebase/firebase-tools/blob/master/src/deploy/functions/runtimes/discovery/v1alpha1.ts

Reply:
and yes agreed about adding a comment